src/vos/vos_obj_cache.c

```diff
@@ -225,19 +225,17 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 			obj->obj_epoch = epoch;
 
 		if (intent == DAOS_INTENT_KILL) {
-			if (!obj->obj_df) /* new object, nothing to delete */
-				D_GOTO(failed, rc = -DER_NONEXIST);
-
 			if (vos_obj_refcount(obj) > 2)
 				D_GOTO(failed, rc = -DER_BUSY);
```

Comment:
is this also buggy? I think we should decref for it...

Reply:
good point.   I should change failed label to do the decref and just not go there if we haven't held the object.